plugins/active-directory/src/active-directory.ts

```diff
@@ -0,0 +1,80 @@
+import { Callback, IPluginAuth, Logger } from '@verdaccio/types';
+import ActiveDirectory from 'activedirectory2';
+
+export const NotAuthMessage = 'AD - Active Directory authentication failed';
+
+export interface ActiveDirectoryConfig {
+  url: string;
+  baseDN: string;
+  domainSuffix: string;
+  groupName?: string | string[];
+}
+
+class ActiveDirectoryPlugin implements IPluginAuth<ActiveDirectoryConfig> {
+  private config: ActiveDirectoryConfig;
+  private logger: Logger;
+
+  public constructor(config: ActiveDirectoryConfig, opts: { logger: Logger }) {
+    this.config = config;
+    this.logger = opts.logger;
+  }
+
+  public authenticate(user: string, password: string, cb: Callback): void {
+    const username = `${user}@${this.config.domainSuffix}`;
+
+    const connectionConfig = {
+      ...this.config,
+      domainSuffix: undefined,
+      username,
+      password,
+    };
+
+    const connection = new ActiveDirectory(connectionConfig);
+
+    connection.authenticate(username, password, (err, isAuthenticated): void => {
+      if (err) {
+        this.logger.warn(`AD - Active Directory authentication failed with error: ${err}`);
+        return cb(err);
+      }
+
+      if (!isAuthenticated) {
+        this.logger.warn(NotAuthMessage);
+        return cb(new Error(NotAuthMessage));
```

Comment:
Here we need to use the VerdaccioError or ErrorCode from commons-api

Reply:
Would it looks like this?
```typescript
return cb(getUnauthorized(NotAuthMessage));
```

Reply:
Added types for all errors @juanpicado